### PR TITLE
Update docs to match running builtins as plugins

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**
Updates docs to match the changes introduced in https://github.com/hashicorp/vault/pull/5536 to run all builtins as plugins.

### CODE REVIEW

File: website/source/api/system/mounts.html.md [hunk 1]
```diff
@@ -108,8 +103,7 @@ This endpoint enables a new secrets engine at the given path.
     These control the default and maximum lease time-to-live, force
     disabling backend caching, and option plugin name for plugin backends
```
Comment (kalafut) [inline website/source/api/system/mounts.html.md]: Remove reference to plugin name & backends.

File: website/source/api/system/plugins-catalog.html.md [hunk 2]
```diff
@@ -9,17 +9,17 @@ description: |-
 
 # `/sys/plugins/catalog`
 
-The `/sys/plugins/catalog` endpoint is used to list, register, update, and
+The `/sys/plugins/catalog` endpoint is used to read, register, update, and
 remove plugins in Vault's catalog. Plugins must be registered before use, and
 once registered backends can use the plugin by querying the catalog.
 
-## List Plugins
+## Read Plugins
```
Comment (kalafut) [inline website/source/api/system/plugins-catalog.html.md]: Even though the Method has changed to GET, I think "List" is still the right word for the heading, and is consistent with following sentence.

File: website/source/guides/operations/plugin-backends.html.md [hunk 3]
```diff
@@ -61,6 +61,14 @@ Put the plugin in the directory:
 $ mv my-mock-plugin /etc/vault/vault_plugins
 ```
 
+Alternatively, if you wanted a custom version of a plugin built into Vault, such as AppRole:
+
+```sh
+$ cd $GOPATH/src/github.com/hashicorp/vault/builtin/credential/approle/cmd/approle
```
Comment (kalafut) [inline website/source/guides/operations/plugin-backends.html.md]:
```suggestion
$ cd $GOPATH/src/github.com/hashicorp/vault/builtin/credential/approle/cmd
```

File: website/source/guides/operations/plugin-backends.html.md [hunk 4]
```diff
@@ -61,6 +61,14 @@ Put the plugin in the directory:
...
+$ go build
```
Comment (kalafut) [inline website/source/guides/operations/plugin-backends.html.md]:
```suggestion
$ go build -o approle
```

File: website/source/api/system/mounts.html.md (hunk 1 above)
Comment (tyrannosaurus-becks) [reply website/source/api/system/mounts.html.md]: Good catch!

File: website/source/api/system/plugins-catalog.html.md (hunk 2 above)
Comment (tyrannosaurus-becks) [reply website/source/api/system/plugins-catalog.html.md]: Good point! Also, revisiting this made me realize I'd not included the new three LIST endpoints for plugins by type, so I'm adding those.

File: website/source/guides/operations/plugin-backends.html.md (hunk 3 above)
Comment (tyrannosaurus-becks) [reply website/source/guides/operations/plugin-backends.html.md]: Arg, I accepted this, but I now realize I shouldn't have. The original directory was correct: https://github.com/hashicorp/vault/tree/master/builtin/credential/approle/cmd/approle

I will push a commit including reverting that string.

File: website/source/guides/operations/plugin-backends.html.md (hunk 4 above)
Comment (tyrannosaurus-becks) [reply website/source/guides/operations/plugin-backends.html.md]: I don't think this is necessary if the cd into the approle directory.

File: website/source/guides/operations/plugin-backends.html.md (hunk 3 above)
Comment (kalafut) [reply website/source/guides/operations/plugin-backends.html.md]: Oops, sorry about that. I happened to have `stable-website` checked out while running through the steps, and it's way old on code.

File: website/source/guides/operations/plugin-backends.html.md (hunk 4 above)
Comment (kalafut) [reply website/source/guides/operations/plugin-backends.html.md]: Yep, this was coupled to my last comment, which is now moot.

**Pull request merged**